src/Symfony/Component/HttpFoundation/UriSigner.php

```diff
@@ -97,30 +136,30 @@ public function check(string $uri): bool
         }
 
         if (empty($params[$this->hashParameter])) {
-            return false;
+            throw new UnSignedUriException($uri, 'The URI is not signed.');
         }
 
         $hash = $params[$this->hashParameter];
         unset($params[$this->hashParameter]);
 
         // In 8.0, remove support for non-url-safe tokens
         if (!hash_equals($this->computeHash($this->buildUrl($url, $params)), strtr(rtrim($hash, '='), ['/' => '_', '+' => '-']))) {
-            return false;
+            throw new UnverifiedSignedUriException($uri, 'The URI signature is invalid.');
```

Comment:
what about moving the message inside the exception class and not making it configurable? I don't think this will ever change. (same for other exceptions)